test/test_model.py

```diff
@@ -65,202 +71,230 @@ def setUp(self):
+        harness.add_relation_unit(relation_id_db1, 'remoteapp1/0')
+        relation_id_db0_b = harness.add_relation('db0', 'another')
+        self.resetBackendCalls(harness)
 
         with pytest.raises(ops.ModelError):
             # You have to specify it by just the integer ID
-            self.model.get_relation('db1', f'db1:{relation_id_db1}')  # type: ignore
-        rel_db1 = self.model.get_relation('db1', relation_id_db1)
+            model.get_relation('db1', f'db1:{relation_id_db1}')  # type: ignore
+        rel_db1 = model.get_relation('db1', relation_id_db1)
         assert isinstance(rel_db1, ops.Relation)
-        self.assertBackendCalls([
+        self.assertBackendCalls(harness, [
             ('relation_ids', 'db1'),
             ('relation_list', relation_id_db1),
         ])
-        dead_rel = self.ensure_relation('db1', 7)
+        dead_rel = self.ensure_relation(model, 'db1', 7)
         assert isinstance(dead_rel, ops.Relation)
-        assert set(dead_rel.data.keys()) == {self.model.unit, self.model.unit.app}
-        assert dead_rel.data[self.model.unit] == {}
-        self.assertBackendCalls([
+        assert set(dead_rel.data.keys()) == {model.unit, model.unit.app}
+        assert dead_rel.data[model.unit] == {}
+        self.assertBackendCalls(harness, [
             ('relation_list', 7),
             ('relation_remote_app_name', 7),
             ('relation_get', 7, 'myapp/0', False),
         ])
 
-        assert self.model.get_relation('db2') is None
-        self.assertBackendCalls([
+        assert model.get_relation('db2') is None
+        self.assertBackendCalls(harness, [
             ('relation_ids', 'db2'),
         ])
-        assert self.model.get_relation('db1') is rel_db1
+        assert model.get_relation('db1') is rel_db1
         with pytest.raises(ops.TooManyRelatedAppsError):
-            self.model.get_relation('db0')
+            model.get_relation('db0')
 
-        self.assertBackendCalls([
+        self.assertBackendCalls(harness, [
             ('relation_ids', 'db0'),
-            ('relation_list', self.relation_id_db0),
+            ('relation_list', relation_id_db0),
             ('relation_remote_app_name', 0),
             ('relation_list', relation_id_db0_b),
             ('relation_remote_app_name', 2),
         ])
 
-    def test_peer_relation_app(self):
-        self.harness.add_relation('db2', 'myapp')
-        rel_dbpeer = self.ensure_relation('db2')
-        assert rel_dbpeer.app is self.model.app
-
-    def test_remote_units_is_our(self):
-        relation_id = self.harness.add_relation('db1', 'remoteapp1')
-        self.harness.add_relation_unit(relation_id, 'remoteapp1/0')
-        self.harness.add_relation_unit(relation_id, 'remoteapp1/1')
-        self.resetBackendCalls()
-
-        for u in self.ensure_relation('db1').units:
+    def test_peer_relation_app(
+        self,
+        harness: ops.testing.Harness[ops.CharmBase],
+        model: ops.Model,
```

Comment:
Another example where this harness isn't adding any value, and I think making it more confusing. We can just remove the model fixture and reference `harness.model` directly in the test.